UndertaleModTool/Editors/UndertaleRoomEditor.xaml.cs

```diff
@@ -232,6 +235,22 @@ private void TreeView_SelectedItemChanged(object sender, RoutedPropertyChangedEv
             {
                 ObjectEditor.Content = obj;
 
+                if (obj is GameObject)
+                {
+                    var room = DataContext as UndertaleRoom;
+                    if (room?.Flags.HasFlag(RoomEntryFlags.IsGMS2) == true)
+                    {
+                        // Check if the selected game object is in the "Game objects (from all layers)" list
+                        var objectItem = GameObjItems.ItemContainerGenerator.ContainerFromItem(obj) as TreeViewItem;
+                        if (objectItem?.IsSelected != true)
+                            isMovable = true;
+                    }
+                    else
+                        isMovable = true;
+                }
+                else
+                    isMovable = movableTypes.Contains(obj.GetType());
```

Comment:
```suggestion
                if (obj is GameObject)
                {
                    var room = DataContext as UndertaleRoom;
                    if (room?.Flags.HasFlag(RoomEntryFlags.IsGMS2))
                    {
                        // Check if the selected game object is in the "Game objects (from all layers)" list
                        var objectItem = GameObjItems.ItemContainerGenerator.ContainerFromItem(obj) as TreeViewItem;
                        if (objectItem?.IsSelected == true)
                            isMovable = false;
                    }
                }
                
                MoveButtonsPanel.IsEnabled = isMovable;
```

Reply:
`room?.Flags.HasFlag(RoomEntryFlags.IsGMS2)` returns a `bool?` type, so it should be `room?.Flags.HasFlag(RoomEntryFlags.IsGMS2) == true` as it was.

Reply:
Ok. Updated